src/collectors/scribe/scribe.py

```diff
@@ -0,0 +1,101 @@
+# coding=utf-8
+
+"""
+A collector for publishing Scribe metrics from a local instance.
+
+#### Dependencies
+
+ * A local install of scribe_ctrl.
+"""
+
+import numbers
+import subprocess
+import diamond.collector
+
+
+class ScribeCollector(diamond.collector.Collector):
+
+    # Mapping of known raw names to counter names
+    COUNTERS = {
+        'denied for queue size': 'denied_queue_size',
+        'denied for rate': 'denied_rate',
+        'lost': 'lost',
+        'received bad': 'received_bad',
+        'received blank category': 'received_blank_category',
+        'received good': 'received_good',
+        'requeue': 'requeue',
+        'retries': 'retries',
+        'sent': 'sent'
+    }
+
+    def get_default_config_help(self):
+        config_help = super(ScribeCollector, self).get_default_config_help()
+        config_help.update({
+        })
+        return config_help
+
+    def get_default_config(self):
+        """
+        Returns the default collector settings
+        """
+        config = super(ScribeCollector, self).get_default_config()
+        config.update({
+            'path': 'scribe',
+            'ports': 1464,
+            'merge_sources': True,
+            'bin': 'scribe_ctrl'
+        })
+        return config
+
+    def _parse_raw_metrics(self, metrics):
+        """
+        Convert raw scribe output to a dictionary of metric names/values.
+        """
+        raw_items = [p for p in metrics.split('\n') if p]
+        data = {}
+        for item in raw_items:
+            source, name, value = [s.strip() for s in item.split(':')]
+            mapped_name = self.COUNTERS.get(name)
+            if mapped_name:
+                data['%s.%s' % (mapped_name, source)] = long(value)
+            else:
+                self.log.warn('Unexpected counter name: %s', name)
+
+        return data
+
+    def _get_metrics(self, port):
+        """
+        Get metrics from a specific port.
+        """
+        command = [self.config['bin'], 'counters', str(port)]
+        try:
+            proc = subprocess.Popen(command, stdout=subprocess.PIPE)
+            out, _ = proc.communicate()
+
+            return self._parse_raw_metrics(out)
+        except Exception:
+            self.log.exception('Unable to collect disk usage.')
```

Comment:
I assume this is from a copy/pasta?  Maybe rename to "Unable to collect scribe metrics"?